src/robotFactory.js

```diff
@@ -1,10 +1,69 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
+    this.coords = {};
+    this.coords.x = position.x ? position.x : 0;
+    this.coords.y = position.y ? position.y : 0;
+  }
+  getInfo() {
+    return `Robot: ${this.name}, `
+    + `Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  };
 
-class FlyingRobot {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
 
-class DeliveryDrone {}
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, position, chipVersion) {
+    super(name, weight, position, chipVersion);
+    this.coords.z = position.z ? position.z : 0;
+  }
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  };
+  goDown(step = 1) {
+    this.coords.z -= step;
+  };
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, position, chipVersion, maxLoadWeight, currentLoad) {
```

Comment:
Later you check if `this.currentLoad === null`. Let's set it as a default value, not to have potential issues with `undefined`:
```suggestion
  constructor(
    name,
    weight,
    position,
    chipVersion,
    maxLoadWeight,
    currentLoad = null,
  ) {
```